### ISSUE TITLE
TASTy file was produced by Scala 3.2.0

### ISSUE THREAD

**Issue by carueda** [outr/scribe#385]
After a few months I'm resuming work on a project that uses scribe 3.9.0. Among other updates I wanted to also bump scribe to a more recent version (even the latest, 3.10.7 at this time). However,  I'm getting the error in the title, more specifically (showing part of my project tests):

```
sbt:root> test
[info] compiling ...
[error] error while loading package$,
[error] class file scribe/package.class is broken, reading aborted with class dotty.tools.tasty.UnpickleException
[error] TASTy signature has wrong version.
[error]  expected: {majorVersion: 28, minorVersion: 1}
[error]  found   : {majorVersion: 28, minorVersion: 2}
[error]
[error] This TASTy file was produced by a more recent, forwards incompatible release.
[error] To read this TASTy file, please upgrade your tooling.
[error] The TASTy file was produced by Scala 3.2.0.
[error] -- [E008] Not Found Error: /Users/carueda/TethysL/tethysl/base/src/main/scala/org/mbari/tethysl/base/butil.scala:53:15
[error] 53 |        scribe.debug(s"FileNotFoundException: $f")
[error]    |        ^^^^^^^^^^^^
[error]    |        value debug is not a member of scribe
...
```

So, it seems scribe now requires Scala 3.2 by the client, and my project uses Scala 3.1.1 and, at the moment, it's not possible to upgrade it to 3.2.0 for other reasons.

By trying things I note that up to scribe 3.10.3, my project builds and runs fine, so the scala 3.2 requirement seems to have been introduced in scribe 3.10.4.

In summary, any comments would be welcome as I plan for future updates involving scribe (of course,  feel free to close this as appropriate). Thanks.

**Comment by darkfrog26** [comment]
@carueda can you clarify why you can't upgrade to 3.2?

**Comment by darkfrog26** [comment]
I was going to do a quick new release just for you that is built on `3.1.1`, but some of the dependencies I'm relying on also are built on `3.2`, so it's not quite so easy. :(

**Comment by carueda** [comment]
Thanks for the quick replies (already useful) and even offer to create a new release!, really appreciated but not need for the trouble! I faced some issues (scribe unrelated) when just setting Scala to 3.2, apparently not straightforward to solve as I would have expected (yes, dealing with scala versions is such a pain). But I'll revisit this more carefully. (My project is not (yet) open source, unfortunately). Thanks again. Fee free to close this ticket.

**Comment by darkfrog26** [comment]
Feel free to follow up here with the issues you're facing. I asked about this on the Scala Discord channel, which is why you'll see a few people watching the comment above about the issues upgrading to 3.2.

**Comment by carueda** [comment]
Thanks @darkfrog26 et al –

fyi I just retried building my project with Scala 3.2.1 (and latest scribe, among other updates). I managed to fix issues in my previous attempts with Scala 3.2, and it seems the only remaining one (for me) is related with [non-local returns](https://docs.scala-lang.org/scala3/reference/dropped-features/nonlocal-returns.html), of which there are several in this project. Too bad this is not just a deprecation warning as indicated in that page:

> Deprecated: Nonlocal Returns
> Returning from nested anonymous functions has been deprecated, and will produce a warning from version 3.2.

but actually already dropped in 3.2! :(

In summary, thanks all again, I'll close this now.  Best regards.

**Comment by carueda** [comment]
Oops, I forgot I had `-Xfatal-warnings` in my scalacOptions! -- I'm now good to proceed (with more time to address that warning).
